### ISSUE TITLE
notificationDisplay event does not work

### ISSUE THREAD

**Issue by michalakis** [OneSignal/OneSignal-Website-SDK#645]
I am using chrome, i put the code as the documentation says in the head

```
 <script src="https://cdn.onesignal.com/sdks/OneSignalSDK.js" async=""></script>
		<script>
		  var OneSignal = window.OneSignal || [];
		  OneSignal.push(function() {
			OneSignal.init({
			  appId: "#########",
			});
			   OneSignal.on('notificationDisplay', function(event) {
    console.warn('OneSignal notification displayed:', event);
  });
			  
			  OneSignal.on('notificationDismiss', function(event) {
    console.warn('OneSignal notification dismissed:', event);
  });
		  });
			
		</script>
```

notificationDismiss doesnt work either for that matter 

is there something i am missing? is there some configuration for this to work?

notifications work i send them through code:

```
// onesignal notification
					$.ajax({
						url: 'https://onesignal.com/api/v1/notifications',
						method: "POST",
						headers: {
							Authorization: 'Basic #######'
						},
						data: {
						  "app_id": #####",
						  "included_segments": ["All"],
						  "data": {"foo": "bar"},
						  "contents": {"en": "message"}
						}, success: function (successMsg)
						{
							console.log(successMsg);
						}, error: function (errorMsg) 
						{
							console.log(errorMsg);
						} 
					});
```

notifications do work, but i do get a console error, dont know if this is related

```
OneSignalError.js:18 Uncaught (in promise) InvalidStateError: The current environment does not support this operation.
    at Function.getServiceWorkerHref (https://cdn.onesignal.com/sdks/OneSignalPageSDKES6.js?v=151001:1:40861)
    at Ae.<anonymous> (https://cdn.onesignal.com/sdks/OneSignalPageSDKES6.js?v=151001:1:139146)
    at Generator.next (<anonymous>)
    at r (https://cdn.onesignal.com/sdks/OneSignalPageSDKES6.js?v=151001:1:716)
```

**Comment by rgomezp** [comment]
Howdy,
You can always send a test notification using `sendSelfNotification` for easier testing.

Can you please provide a bit more details? What browser are you seeing this in? Can you reproduce on others? Also what is your operating system?

Cheers

**Comment by rgomezp** [comment]
Closing due to no response

**Comment by markelliottopoc** [comment]
HI, I have the same issue. Have tested in Firefox, Chrome and MS Edge with no success. I've tested with sendSelfNotification  accoridng to your suggestion but still don't see the event firing.

I'm registering as so:

        OneSignal.push(function () {

                        OneSignal.sendTag("UserId", "@User.Identity.Name")
                        .then(function (tagsSent) {
                            // Callback called when tags have finished sending
                            console.log('sendTag then', tagsSent);
                        });

                        OneSignal.once('notificationDisplay', function (event) {
                            console.log('OneSignal notification displayed:', event);
                        });

                        OneSignal.once('notificationDismiss', function (event) {
                            console.log('OneSignal notification dismissed:', event);
                        });
                    });

**Comment by sajideen-maker** [comment]
Hi, I am facing the same issue. Firstly I tried the npm package of react-oneSignal but this cant fulfills my requirement. I want to update data on my page when I receive a notification. I have tried all events from web-SDK but nothing works for me. it would be great if you tell me a specific event that I need to trigger. I am using 'notificationDisplay' but this is not working in my scenario 

`OneSignal.push(["init", {
  appId: process.env.REACT_APP_ONE_SIGNAL_ID,
  allowLocalhostAsSecureOrigin: true,
}]);
OneSignal.push(function() {
  OneSignal.showSlidedownPrompt();
});
OneSignal.push(function() {
  OneSignal.setExternalUserId(id);
});
OneSignal.push(function() {
  OneSignal.on('notificationDisplay', function(event) {
    console.log('notificationDisplay');
    console.log('event', event);
  });
});`
